Sort task history applying most recent first

JorgeMartinezG opened the pull request:
closes #3044

ramyaragupathy commented on backend/models/postgis/task.py:
```diff
@@ -1013,7 +1013,7 @@ def as_dto(
         """Get dto with any task instructions"""
         task_history = []
-        for action in reversed(self.task_history):
+        for action in self.task_history:
```
Ran this against a task #951 on project 5660, comparing task history ID values from production and current branch:

Production results | Current branch results
-------------------|-----------------------
7683258<br>7683259<br>7683260<br>7683261<br>7717936<br>7717508<br>10858873<br>10858874<br>10858872<br>11187402|11187402<br>10858872<br>10858874<br>10858873<br>7717508<br>7717936<br>7683261<br>7683260<br>7683259<br>7683258

The current branch simply inverts the array list. If you see the production results you can observe there is already a problem in task history ordering which is carried to the array reversal function too. We should correct this ordering in [Task.get](https://github.com/hotosm/tasking-manager/blob/develop/backend/models/postgis/task.py#L573) - but if this affects other methods, then may be we should go for sort with python instead of reversed method.  I tired `sorted(self.task_history, key=lambda x:x.id, reverse=True)` and this returned me IDs in the following order and it aligned well with the `action_date` timestamps.

11187402<br>10858874<br>10858873<br>10858872<br>7717936<br>7717508<br>7683261<br>7683260<br>7683259<br>7683258

JorgeMartinezG commented:
@ramyaragupathy Instead of performing the sort in Python, I modified the relationship in order to perform the sort directly in Postgres. Please pull changes again since I rebased the branch
